hibernate-core/src/main/java/org/hibernate/dialect/DerbyDialect.java

```diff
@@ -252,6 +262,12 @@ public String castPattern(CastType from, CastType to) {
 					return "case ?1 when false then 0 when true then 1 end";
 				}
 				break;
+			case STRING:
+				// See https://issues.apache.org/jira/browse/DERBY-2072
+				if ( from.getKind() == CastTypeKind.NUMERIC ) {
+					return "cast(cast(?1 as char(38)) as ?2)";
```

Comment:
38?

Is there something magical about this number?

Reply:
Oh, right, I guess it's the maximum number of digits in a 32 bit integer....

Reply:
Rmm, nope, it's not even the max number of digits in a long.

Reply:
Curious to know where 38 came from.

Reply:
This was the suggested workaround in the comment https://issues.apache.org/jira/browse/DERBY-2072?focusedCommentId=17154473&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-17154473

I don't know what the maximum decimal precision is, so we might just as well use 254(char max capacity) to be safe, but 32 and 64 bit values should definitely fit into that data type. I'll just use the maximum capacity then and document this.

Reply:
> I don't know what the maximum decimal precision is

I think it's 31, but I might be wrong.